app/helpers/templates_helper.rb

```diff
@@ -58,27 +58,29 @@ def hide_resource_type_input(obj)
     'hide' unless obj.value_type == 'search'
   end
 
-  def mount_report_template_input(input_value)
-    return if input_value.nil?
-
+  def template_input_f(f, options = {})
+    input_value = f.object
     input = input_value.template_input
-    controller = input.resource_type&.tableize
 
-    react_component('TemplateInput', { data: {
-                      value: input_value.value.to_s,
-                      required: input.required,
-                      template: 'report_template_report',
-                      description: input.description,
-                      supportedTypes: TemplateInput::VALUE_TYPE,
-                      resourceType: controller,
-                      id: input.id,
-                      useKeyShortcuts: false,
-                      url: search_path(controller),
-                      label: input.name,
-                      type: input.value_type,
-                      initialError: input_value.errors[:value].join("\n").presence,
-                      resourceTypes: Hash[Permission.resources.map { |d| [d.tableize.to_s, d] }],
-                    },
-    })
+    options.reverse_merge!(label: input.name, id: input.name, label_help: input.description, required: input.required)
+
+    if input.options.present?
+      selectable_f(f, :value, input.options_array, { include_blank: !input.required }, id: input.name, onchange: options[:onchange])
+    elsif input.value_type == 'plain' || input.value_type.nil?
+      textarea_f(f, :value, rows: 2, onchange: options[:onchange], id: input.name, class: input.hidden_value? ? 'masked-input' : '')
+    else
+      input_type = input.value_type
+      input_type = 'dateTime' if input_type == 'date'
```

Comment:
I've dropped the input value type renaming to separate PR (#8069) so I'm doing this instead, let's do it one by one :+1: